GitUpKit/Core/GCHistory.m

```diff
@@ -1242,17 +1278,20 @@ - (NSArray*)lookupCommitsForFile:(NSString*)path followRenames:(BOOL)follow erro
                 if (status == GIT_OK) {
                   git_diff* diff = NULL;
                   status = git_diff_tree_to_tree(&diff, self.private, parentTree, tree, &diffOptions);
-                  if ((status == GIT_OK) && follow) {
+                  if ((status == GIT_OK) && theOptions.followRenames) {
                     status = git_diff_find_similar(diff, &findOptions);
                   }
                   if (status == GIT_OK) {
                     for (size_t i2 = 0, count2 = git_diff_num_deltas(diff); i2 < count2; ++i2) {
                       const git_diff_delta* delta = git_diff_get_delta(diff, i2);
                       if (strcmp(delta->new_file.path, fileName) == 0) {
-                        GCCommit* newCommit = [[GCCommit alloc] initWithRepository:self commit:commit];
+                        // here we need to retain commit.
+                        git_commit *copy_commit;
+                        git_commit_dup(&copy_commit, commit);
+                        GCCommit* newCommit = [[GCCommit alloc] initWithRepository:self commit:copy_commit];
                         [commits addObject:newCommit];
                         [newCommit release];
-                        commit = NULL;
+                        //                        commit = NULL; // well, if we are here, we should terminate for-loop (count = git_commit_parentcount(commit))
```

Comment:
Don't leave commented out code. It should either be there with some action to take in the future or it shouldn't. There's no way to take further action on this.